tests/unit/aggregation/_inputs.py

```diff
@@ -2,138 +2,76 @@
+    ``min(n_cols, n_rows - 1)``, such that there exists a vector ``0<v`` with ``v @ matrix = 0``.
     """
 
-    if orthogonal_matrix.shape[1] > n_cols:
-        raise ValueError(
-            f"Parameter `n_cols` should exceed the second dimension of the provided matrix. Found "
-            f"`n_cols = {n_cols}` and `partial_matrix.shape[1] = {orthogonal_matrix.shape[1]}`."
-        )
-
-    for i in range(n_cols - 1):
-        orthogonal_matrix = _augment_orthogonal_matrix(orthogonal_matrix)
-    return orthogonal_matrix
-
+    v = torch.abs(torch.randn([n_rows]))
+    return _generate_matrix_with_orthogonal_vector(v, n_cols)
 
-def _generate_unitary_matrix(n_rows: int, n_cols: int) -> Tensor:
-    """Generates a unitary matrix of shape [n_rows, n_cols]."""
 
-    _check_valid_dimensions(n_rows, n_cols)
-    partial_matrix = torch.randn([n_rows, 1])
-    partial_matrix = torch.nn.functional.normalize(partial_matrix, dim=0)
-
-    unitary_matrix = _complete_orthogonal_matrix(partial_matrix, n_cols)
-    return unitary_matrix
-
-
-def _generate_unitary_matrix_with_positive_column(n_rows: int, n_cols: int) -> Tensor:
+def _generate_weak_stationary_matrix(n_rows: int, n_cols: int) -> Tensor:
     """
-    Generates a unitary matrix of shape [n_rows, n_cols] with the first column consisting of an all
-    positive vector.
+    Generates a random matrix of shape [``n_rows``, ``n_cols``] with rank
+    ``min(n_cols, n_rows - 1)``, such that there exists a vector ``0<=v`` with at least one
+    coordinate equal to ``0`` and such that ``v @ matrix = 0``.
     """
-    _check_valid_dimensions(n_rows, n_cols)
-    partial_matrix = torch.abs(torch.randn([n_rows, 1]))
-    partial_matrix = torch.nn.functional.normalize(partial_matrix, dim=0)
 
-    unitary_matrix_with_positive_column = _complete_orthogonal_matrix(partial_matrix, n_cols)
-    return unitary_matrix_with_positive_column
+    v = torch.abs(torch.randn([n_rows]))
+    v[torch.randint(0, n_rows, [])] = 0.0
+    return _generate_matrix_with_orthogonal_vector(v, n_cols)
 
 
-def _generate_diagonal_singular_values(rank: int) -> Tensor:
-    """
-    generates a diagonal matrix of positive values sorted in descending order.
-    """
-    singular_values = torch.abs(torch.randn([rank]))
-    singular_values = torch.sort(singular_values, descending=True)[0]
-    S = torch.diag(singular_values)
-    return S
+def _generate_orthogonal_matrix(dim: int) -> Tensor:
+    """Uniformly generates a random orthogonal matrix of shape [``dim``, ``dim``]."""
 
+    A = torch.randn([dim, dim])
+    Q, _ = torch.linalg.qr(A)
+    return Q
 
-def generate_matrix(n_rows: int, n_cols: int, rank: int) -> Tensor:
+
+def _generate_matrix_with_orthogonal_vector(vector: Tensor, n_cols: int) -> Tensor:
     """
-    Generates a random matrix of shape [``n_rows``, ``n_cols``] with provided ``rank``.
+    Generates a random matrix of shape [``len(vector)``, ``n_cols``] with rank
```

Comment:
Generates a random unitary matrix uniformly over all unitary matrices of shape ... with rank ... such that ...

Reply:
Or "Uniformly generates a random unitary matrix of shape [...] with rank [...] such that [...]."
To match the docstring of the previous one?